components/atom/backToTop/src/index.js

```diff
@@ -16,95 +16,83 @@ const STYLES = {
   LIGHT: 'light'
 }
 
-class AtomBackToTop extends PureComponent {
-  state = {
-    show: null,
-    hover: false
-  }
-
-  intervalId = 0
-  container = null
-
-  scrollStep = () => {
-    const {scrollTop} = this.container
-    const {scrollSteps} = this.props
-    const {intervalId} = this
+const AtomBackToTop = ({
+  iconTop: IconTop,
+  textTop,
+  scrollSteps,
+  scrollIntervalTime,
+  refContainer,
+  style
+}) => {
+  const [show, setShow] = useState(null)
+  const [hover, setHover] = useState(false)
+
+  let intervalId = 0
+  let container = useRef(null)
+
+  container = getTarget(refContainer)
+  const {scrollHeight, clientHeight} = container
+
+  useEffect(
+    () => {
+      const halfHeight = Math.floor((scrollHeight - clientHeight) / 2)
+
+      const handleScroll = () => {
+        const {scrollTop} = container
+
+        if (scrollTop > halfHeight) {
+          if (!show) setShow(true)
+        } else {
+          if (show) {
+            setShow(false)
+            setHover(false)
+          }
+        }
+      }
+      container.addEventListener('scroll', handleScroll)
+      return () => {
+        container.removeEventListener('scroll', handleScroll)
+      }
+    },
+    [clientHeight, intervalId, scrollHeight, show]
+  )
+
+  const scrollStep = () => {
+    const {scrollTop} = container
```

Comment:
I would say you have to extract the value from `container.current`.

Reply:
The `getTarget` method used to get the DOM container is already returning the `target.current`